atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/PreCommitConditions.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.transaction.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.palantir.atlasdb.transaction.api.PreCommitCondition;
+
+public final class PreCommitConditions {
+
+    private static final Logger log = LoggerFactory.getLogger(PreCommitConditions.class);
+
+    public static final PreCommitCondition NO_OP = new PreCommitCondition() {
+        @Override
+        public void throwIfConditionInvalid(long timestamp) {}
+
+        @Override
+        public void cleanup() {}
+    };
+
+    private PreCommitConditions() {}
+
+    /**
+     * Runs a cleanup task as best-effort, since we want to avoid throwing during cleanup.
+     */
+    public static void runCleanupTask(Runnable task) {
+        try {
+            task.run();
+        } catch (Throwable t) {
```

Comment:
Not 100% sure if we want to swallow `Error`s - should this just be `Exception`?

Reply:
`Exception` should be fine, I'll switch to that and merge.